blocks/productstable/productstable.js

```diff
@@ -0,0 +1,84 @@
+import {
+  label, tr, th, td,
+} from '../../scripts/dom-helpers.js';
+
+export default async function decorate(block) {
+  const table = document.createElement('table');
+  const thead = document.createElement('thead');
+  const tbody = document.createElement('tbody');
+
+  const header = !block.classList.contains('no-header');
+  if (header) table.append(thead);
+  table.append(tbody);
+
+  [...block.children].forEach((child) => {
+    [...child.children].forEach((col) => {
+      // const row = document.createElement('tr');
+      const row = tr();
+      col.querySelectorAll('div.heading').forEach((div) => {
+        const cell = th();
+        cell.setAttribute('scope', 'col');
+        cell.innerHTML = div.innerHTML;
+        row.append(cell);
+        thead.append(row);
+      });
+      col.querySelectorAll('div.data').forEach((div) => {
+        const cell = td();
+        cell.innerHTML = div.innerHTML;
+        row.append(cell);
+        tbody.append(row);
+      });
+    });
+  });
+
+  // Addition of show more row
+  const showMoreRow = tr({ class: 'showmore' });
+  const showMoreCell = td();
+  showMoreCell.setAttribute('colspan', '100%');
```

Comment:
You can leverage dom-helpers here!  36 & 37 can be combined into 1
`const showMoreCell = td({ colspan: '100%' });`